README.md

```diff
@@ -295,6 +296,48 @@ For new snippets the general implementation steps are:
   }
 ```
 
+### Zip a directory
+
+```java
+  public static void zipDirectory (String srcDirectoryName, String zipFileName) throws IOException {
+    var srcDirectory = new File(srcDirectoryName);
+    try (
+      var fileOut = new FileOutputStream(zipFileName);
+      var zipOut = new ZipOutputStream(fileOut);
+    ) {
+      zipFile(srcDirectory, srcDirectory.getName(), zipOut);
+    }
+  }
+  public static void zipFile(File fileToZip, String fileName, ZipOutputStream zipOut) throws IOException{
+    if (fileToZip.isHidden()) { 
+      return;
+    }
+    if (fileToZip.isDirectory()) {
+       if (fileName.endsWith("/")) {
+          zipOut.putNextEntry(new ZipEntry(fileName)); 
+           zipOut.closeEntry();
+       } else {
+         zipOut.putNextEntry(new ZipEntry(fileName + "/"));
+         zipOut.closeEntry();
+       }
+       var children = fileToZip.listFiles();
```

Comment:
There's a possibility that `listFiles` returns null when IO error occurs, so we should check for that

Reply:
yup, that's really common

Reply:
The function already throws IOException. Does it make sense to explicitly check for this error?
If an internal I/O error happens, the zipping should stop in my opinion as if we handle this error, the end result might be partial and incorrect zipping.

Reply:
We should explicitly check and catch an exception, so that we can handle the necessary action, just like what you said, say the end result might be partial and incorrect zipping, instead of simply leaving it just like that,  by checking and catching it, you can correct the error, just like repeating the process again so that it will be corrected or a code to check what caused the error.